### PR TITLE
LPD-37970 Add Publication Timeline History support for Blogs

### PR DESCRIPTION
**Pull request by ctang3**
https://liferay.atlassian.net/browse/LPD-37970

### CODE REVIEW

**Comment by ctang3** [comment]
ci:test:publications

**Comment by ctang3** [comment]
ci:test:relevant

**Comment by david-truong** [comment]
ci:test:sf

**Comment by david-truong** [comment]
ci:test:publications

**Comment by david-truong** [comment]
ci:test:relevant

**Comment by david-truong** [comment]
ci:test:relevant

**Comment by david-truong** [comment]
ci:test:publications

File: modules/apps/blogs/blogs-web/src/main/java/com/liferay/blogs/web/internal/portlet/BlogsPortlet.java [hunk 1]
```diff
@@ -84,6 +88,13 @@ protected void doDispatch(
 				_portal.getLiferayPortletRequest(renderRequest),
 				_portal.getLiferayPortletResponse(renderResponse)));
 
+		HttpServletRequest httpServletRequest = _portal.getHttpServletRequest(
+			renderRequest);
+
+		httpServletRequest.setAttribute(
+			CTTimelineKeys.CLASS_NAME, BlogsEntry.class.getName());
+		httpServletRequest.setAttribute(CTTimelineKeys.CLASS_PK, null);
```
Comment (david-truong) [inline modules/apps/blogs/blogs-web/src/main/java/com/liferay/blogs/web/internal/portlet/BlogsPortlet.java:96]: @ctang3 do we need to set CLASS_PK if it is null

Comment (ctang3) [reply modules/apps/blogs/blogs-web/src/main/java/com/liferay/blogs/web/internal/portlet/BlogsPortlet.java:96]: Yes - without manually setting it to `null` here, it was being set to `0` elsewhere. This makes the timeline icon not appear. I believe this is also the same issue that Gi mentioned during yesterday's standup too.

Comment (david-truong) [reply modules/apps/blogs/blogs-web/src/main/java/com/liferay/blogs/web/internal/portlet/BlogsPortlet.java:96]: can you figure out where its being set to 0 and we should fix that instead?

Comment (gislayne-vitorino) [reply modules/apps/blogs/blogs-web/src/main/java/com/liferay/blogs/web/internal/portlet/BlogsPortlet.java:96]: Hey guys, I did it like this https://github.com/liferay-site-management/liferay-portal/pull/1501/commits/33dfbb8374f12326493a4eecd5041d0d2c5e02ea, 

I noticed that some entities, like ddmStruture and ddmTemplate were returning empty responses on the api when we passed 0 as the classPK. JournalArticles worked fine with adding the `&classPK=0` at the end of the query, but some other didn't. 
Doing this I just passed the class_PK when it was necessary, like when we want to view/edit a specific entry
cc: @ctang3 @david-truong

**Comment by david-truong** [comment]
@ctang3 you can just remove the nulls and we can forward as it is.  We don't need to wait for Gi's pr

**Comment by ctang3** [comment]
ci:test:sf

**Comment by ctang3** [comment]
ci:forward

**Pull request closed without merging**